library/src/main/java/org/xmtp/android/library/ApiClient.kt

```diff
@@ -0,0 +1,66 @@
+package org.xmtp.android.library
+
+import androidx.annotation.VisibleForTesting
+import io.grpc.*
+import org.xmtp.android.library.messages.Topic
+import org.xmtp.proto.message.api.v1.MessageApiGrpcKt
+import org.xmtp.proto.message.api.v1.MessageApiOuterClass.*
+import java.io.Closeable
+import java.util.concurrent.TimeUnit
+
+data class ApiClient(val environment: XMTPEnvironment, val secure: Boolean = true) : Closeable {
+    companion object {
+        @VisibleForTesting
+        val AUTHORIZATION_HEADER_KEY: Metadata.Key<String> =
+            Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER)
+
+        @VisibleForTesting
+        val CLIENT_VERSION_HEADER_KEY: Metadata.Key<String> =
+            Metadata.Key.of("X-Client-Version", Metadata.ASCII_STRING_MARSHALLER)
+
+        @VisibleForTesting
+        val APP_VERSION_HEADER_KEY: Metadata.Key<String> =
+            Metadata.Key.of("X-App-Version", Metadata.ASCII_STRING_MARSHALLER)
+    }
+
+    private val channel: ManagedChannel = Grpc.newChannelBuilderForAddress(
+        environment.rawValue,
+        5556,
+        if (secure) {
+            TlsChannelCredentials.create()
+        } else {
+            InsecureChannelCredentials.create()
+        }
+    ).build()
+
+    private val client: MessageApiGrpcKt.MessageApiCoroutineStub =
+        MessageApiGrpcKt.MessageApiCoroutineStub(channel)
+    private var authToken: String? = null
+
+    fun setAuthToken(token: String): String {
+        authToken = token
+        return authToken as String
+    }
+
+    suspend fun query(topics: List<Topic>): QueryResponse {
+        val request = QueryRequest.newBuilder()
+            .addAllContentTopics(topics.map { it.description }).build()
+
+        val headers = Metadata()
+        headers.put(AUTHORIZATION_HEADER_KEY, "Bearer $authToken")
```

Comment:
With the nullable value, I was thinking we could do something like this. I figure it's better to error bc unauth'd then passing a null / blank value.

```suggestion
        authToken?.let { token ->
            headers.put(AUTHORIZATION_HEADER_KEY, "Bearer $token")
        }
 ```